src/HaplotypingUtils.cpp

```diff
@@ -87,22 +87,22 @@ void GeneticHaplotyper::Apply(Pedigree * & ped, double Rsq, const char * logname
 	data.resize(0);
 
 	if (__chrom == "X") ped->chromosomeX = true;
-	std::string a_freq_output = "-allele-freqs.log";
-	a_freq_output = logname+a_freq_output;
-	const char *a_freq_out = a_freq_output.c_str();
-	MarkerCluster::EstimateAlleleFrequencies(*ped,a_freq_out,reorder);
-	for (int i = 0; i < ped->markerCount; i++)
-	{
-		MarkerInfo * info = ped->GetMarkerInfo(i);
-		std::cout<<ped->markerNames[i]<<": ";
-		for (int j=0; j<info->freq.Length(); j++)
-			std::cout<<info->freq[j]<<" ";
-		std::cout<<"\ntotal familyCount:"<<ped->familyCount<<std::endl;
-	}
+	//std::string a_freq_output = "-allele-freqs.log";
```

Comment:
Because these lines will cause kernel died in some cases due to failing to phase haplotypes. So I tried to comment them and use the functions in your old version. They don't have kernel died issue. If failing to phase haplotypes, they return empty list.
But I do have a question about what is the difference between the two `EstimateFrequencies` functions.

Reply:
Oh so these lines are added by @percylinhai ?